components/controls/notifications/NotificationSettings.tsx

```diff
@@ -60,31 +119,26 @@ const NotificationSettings = () => {
             key={setting.name}
             notificationName={setting.name}
             value={setting.value}
-            onValueChange={() =>
-              saveNotifications({
-                ...notificationSettings,
-                [setting.name]: !setting.value,
-              })
-            }
+            onValueChange={() => saveNotifications({ [setting.name]: !setting.value })}
           />
         ))}
       </Field>
 
-      <Field label={t('emailNotifications')}>
-        {emailNotifications.map((setting) => (
-          <NotificationControl
-            key={setting.name}
-            notificationName={setting.name}
-            value={setting.value}
-            onValueChange={() =>
-              saveNotifications({
-                ...notificationSettings,
-                [setting.name]: !setting.value,
-              })
-            }
```

Comment:
The switches sadly have the same behavior as before, switching for a brief moment, then switching back, before finally switching to the desired position, I think this is a problem with the Switch itself, that it expects to be always switched after a press. The solution would be to use a state for the switches that updates instantly, because the mutation seems to be a little bit too slow, or change it for a custom Switch, which would not be difficult